File: peek_plugin_search/_private/tuples/EncodedSearchIndexChunkTuple.py
from datetime import datetime

from peek_plugin_search._private.PluginNames import searchTuplePrefix
from vortex.Tuple import Tuple, addTupleType, TupleField


@addTupleType
class EncodedSearchIndexChunkTuple(Tuple):
    """ Search Result Detail Tuple

    This tuple represents the details of a search result.
    """
    __tupleType__ = searchTuplePrefix + 'EncodedSearchIndexChunkTuple'

    #:  The name of the detail
    chunkKey: str = TupleField()

    #:  The value of the detail
    lastUpdate: datetime = TupleField()

    #:  The encoded payload containing the :code:`SearchIndexChunk`
    encodedPayload: bytes = TupleField()

File: peek_plugin_search/plugin-module/_private/search-index-loader/EncodedSearchIndexChunkTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {searchTuplePrefix} from "../PluginNames";


@addTupleType
export class EncodedSearchIndexChunkTuple extends Tuple {
    public static readonly tupleName = searchTuplePrefix + "EncodedSearchIndexChunkTuple";

    chunkKey: string;
    lastUpdate: string;
    encodedPayload: string;

    constructor() {
        super(EncodedSearchIndexChunkTuple.tupleName)
    }
}
